python-checks/src/main/java/org/sonar/python/checks/BuiltinShadowingAssignmentCheck.java

```diff
@@ -120,7 +124,18 @@ private void checkAnnotatedAssignment(SubscriptionContext ctx) {
       Expression variable = assignment.variable();
       Token equalToken = assignment.equalToken();
       if (equalToken != null && shouldReportIssue(variable)) {
-        ctx.addIssue(variable, String.format(MESSAGE, VRBL_ISSUE_TYPE));
+        raiseIssueForVariable(ctx, (Name) variable);
+      }
+    }
+  }
+
+  private void raiseIssueForVariable(SubscriptionContext ctx, Name variable) {
+    if (variable.symbol() != null) {
```

Comment:
Nitpick: I think you can assign `variableSymbol` before so you don't have to access it twice.

Reply:
Actually, this was a way to overcome issue with QG. It's a defensive check, but if I will move an assignment a line above -I will have to merge if clauses...

Reply:
Found a way to overcome this